Add `z.string().boolean()` function to parse a boolean value represented by string logically (#2989)
colinhacks/zod: README.md

ipv6.parse("192.168.1.1"); // fail
```

### Logical Boolean Parsing

> The `z.string().boolean()` method transform a string value representing a boolean logically.

Reviewer: ```suggestion
The `z.string().boolean()` method transforms a string value representing a boolean logically.
```

It accepts `"1"`, `"t"`, `"T"`, `"TRUE"`, `"true"`, `"True"` for `true`, `"0"`, `"f"`, `"F"`, `"FALSE"`, `"false"`, `"False"` for `false`.

> > This function's behavior does not along with Typescript standard `Boolean()` function.
> >
> > If you expect the behavior, use [`z.coerce.boolean()`](#coercion-for-primitives) instead.

Reviewer: ```suggestion
> This method does not reproduce [`Boolean()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean)'s behavior.
>
> If you expect [`Boolean()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean)'s behavior, use [`z.coerce.boolean()`](#coercion-for-primitives) instead.
```

```ts
const b = z.string().boolean();

b.parse("1"); // true
b.parse("0"); // false
b.parse("true"); // true
b.parse("false"); // false
b.parse("True"); // true
b.parse("False"); // false

b.parse("other"); // fail
b.parse(undefined); // fail
b.parse(null); // fail
```

> You can additionally use the `catch` function, which handle a parsing error as `true` or `false`.

Reviewer: ```suggestion
You can additionally use the `catch` function, which will return `true` or `false` whenever a parsing error occurs.
```

```ts
const b = z.string().boolean().catch(false);

b.parse("other"); // false
b.parse(undefined); // false
b.parse(null); // false
```

> Also you can use `default()` to handle only `undefined` value.

Reviewer: ```suggestion
You can also use `default()` to fallback to a value on the absence of an input.
```
